src/firecracker/src/main.rs

```
@@ -485,12 +520,14 @@ fn build_microvm_from_json(
     boot_timer_enabled: bool,
     mmds_size_limit: usize,
     metadata_json: Option<&str>,
-) -> std::result::Result<(VmResources, Arc<Mutex<vmm::Vmm>>), FcExitCode> {
+) -> Result<(VmResources, Arc<Mutex<vmm::Vmm>>), FcExitError> {
```

This should return a custom error type (same comment as on `run_with_api`).